smithy-aws-iam-traits/src/main/resources/META-INF/smithy/aws.iam.smithy

```diff
@@ -131,6 +198,9 @@ enum ConditionKeyType {
 @private
 string IamIdentifier
 
+@private
+string ResourceName
```

Comment:
Should an Resource name be an Idref?

Reply:
No, as it contains only the `Shape Name` portion of a `ShapeId` and must not be an absolute Id as required by `idRef`.